modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -494,8 +496,17 @@ TopLevelElement createTypeRef(DslElementSyntax typeDsl, ObjectType type, boolean
     return createRefElement(refQName, isRequired);
   }
 
-  MetadataTypeVisitor getParameterDeclarationVisitor(final ExtensionType extensionType, final ExplicitGroup all,
-                                                     final ParameterModel parameterModel) {
+  void declareAsParameter(MetadataType type, ExtensionType extensionType, ExplicitGroup all, ParameterModel parameterModel) {
+    if (ExtensionModelUtils.isContent(parameterModel) || TypeUtils.isContent(type)) {
```

Comment:
no. when parsing top level elements, a synthetic parameterModel is used temporarily in order to reuse code and keep it simpler. It's just an implementation detail